core/hybrisadaptor.cpp

```diff
@@ -209,11 +209,11 @@ HybrisManager::HybrisManager(QObject *parent)
     }
 
     /* Open android sensor device */
-#ifdef SENSORS_DEVICE_API_VERSION_1_0
-    err = sensors_open_1(&m_halModule->common, &m_halDevice);
-#else
-    err = sensors_open(&m_halModule->common, &m_halDevice);
-#endif
+    if (m_halDevice->common.version >= SENSORS_DEVICE_API_VERSION_1_0)
```

Comment:
m_halDevice is initialized to null. Then it is dereferenced to decide how to make it non-null -> this can't be right / what am I missing?